### PR TITLE
Try to decode JSON the request

### PR DESCRIPTION
**Pull request by amitaibu**
lessons learned from [ng-admin](https://github.com/Gizra/restful/issues/191)

Also, ng-admin sends `id: null` on POST. @mateu-aguilo-bosch apart of unset of the `$request['id']` can you think of a nicer way?

### CODE REVIEW

**Comment by e0ipso** [comment]
Are we supporting JSON encoded input now?

Handling a particular use case for ng-admin makes me uneasy. Also what if someone expects to have the `'id'` field populated?. My take is that for this cases the implementor module is in charge of implementing [`hook_restful_parse_request_alter`](https://github.com/Gizra/restful/blob/7.x-1.x/restful.api.php#L14-L24); to remove whatever they need.

**Comment by amitaibu** [comment]
> Are we supporting JSON encoded input now?

That's actually non ng-admin specific, but more AngularJs specific. Seems that `$http` doesn't send the params as you would expect (so usually I see [this](http://stackoverflow.com/a/22807867/750039) workaround being used). So the `drupal_json_decode()` is just to be a little nicer to Angular, without a big performance penalty. 

> My take is that for this cases the implementor module is in charge of implementing hook_restful_parse_request_alter; to remove whatever they need.

I agree, that's indeed a better idea.

**Comment by e0ipso** [comment]
> but more AngularJs specific

Well, we are supporting the PHP specific urlencoded crap in that gets turned into the $_POST superglobal… Anyways, you can do that also in `hook_restful_parse_request_alter` but I'm OK on providing this as a default since I cannot see a real use case for payloads in XML or HAL or proprietary formats (excel? xD).

File: restful.module [hunk 1]
```diff
@@ -697,11 +697,16 @@ function restful_parse_request() {
   }
 
   if (!$request && $query_string = file_get_contents('php://input')) {
-    // When trying to POST using curl on simpleTest it doesn't reach
-    // $_POST, so we try to re-grab it here.
-    parse_str($query_string, $request);
-  }
 
+    if ($decoded_json = drupal_json_decode($query_string)) {
```
Comment (e0ipso) [inline restful.module:703]: We can keep this part.

File: plugins/restful/RestfulBase.php [hunk 2]
```diff
@@ -131,6 +131,9 @@ public function setRequest(array $request = array()) {
    */
   public static function cleanRequest(&$request) {
     unset($request['__application']);
+
+    // @todo: How to get this in a clean way?
+    unset($request['id']);
```
Comment (e0ipso) [inline plugins/restful/RestfulBase.php]: Let's remove this.

**Comment by amitaibu** [comment]
Ready for review.

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch ^^

**Pull request merged**